mypy_django_plugin/django/context.py

```diff
@@ -364,13 +366,19 @@ def _resolve_field_from_parts(
 
             field = currently_observed_model._meta.get_field(field_part)
             if isinstance(field, RelatedField):
-                currently_observed_model = field.related_model
+                related_model = self.get_field_related_model_cls(field)
```

Comment:
The code duplication is quite high. Can you please refactor it to be a method / function?